index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java

```diff
@@ -121,16 +88,42 @@ private static List<URI> getAffiliations(RequestInfo requestInfo, URI topLevelOr
         }
     }
 
+    private static List<URI> extractQueryParamAffiliations(RequestInfo requestInfo) {
+        return requestInfo.getQueryParameterOpt(QUERY_PARAM_AFFILIATIONS)
+                   .map(s -> Arrays.stream(s.split(",")).map(URI::create).collect(Collectors.toList()))
+                   .orElse(null);
+    }
+
+    @JacocoGenerated
+    private static AuthorizedBackendUriRetriever defaultUriRetriever() {
+        return new AuthorizedBackendUriRetriever(new Environment().readEnv("BACKEND_CLIENT_AUTH_URL"),
+                                                 new Environment().readEnv("BACKEND_CLIENT_SECRET_NAME"));
+    }
+
+    private CandidateSearchParameters getCandidateSearchParameters(RequestInfo requestInfo)
+        throws UnauthorizedException {
+        var isAdmin = userIsNviAdmin(requestInfo);
+        var topLevelOrg = !isAdmin ? requestInfo.getTopLevelOrgCristinId().orElseThrow() : null;
```

Comment:
not keen on the negation

Reply:
There is something about this code that seems like the order of operations is wrong, but maybe I'm wrong